syntax.php

```diff
@@ -67,438 +62,22 @@ function connectTo($mode) {
-     * If $count is 0, then $haystack is returned without modification.
-     *
-     * @param string   $needle   substring to replace
-     * @param string[] $replace  a numerically indexed array of substitutions for $needle
-     * @param int      $count    number of $needles to be replaced by the same element of $replace
-     * @param string   $haystack string to be searched
-     * @return string  $haystack with the substitution applied
-     */
-    static function str_replace_array($needle, $replace, $count, $haystack) {
-        if ($count <= 0) {
-            return $haystack;
-        }
-        $haystackArray = explode($needle, $haystack);
-        $res = '';
-        foreach ($haystackArray as $i => $piece) {
-            $res .= $piece;
-            // "(int) ($i / $count)" simulates integer division.
-            $replaceIndex = (int) ($i / $count);
-            // Note that $replaceIndex will be out of bounds in $replace for the last loop iteration.
-            // In that case, the array access yields NULL, which is interpreted as the empty string.
-            // This is what we need, because there was no $needle after the last $piece.
-            $res .= $replace[$replaceIndex];
-        }
-        return $res;
-    }
-
-    /**
-     * Removes all occurrences of $needle in all strings in the array $haystack recursively.
-     *
-     * @param string $needle   substring or array of substrings to remove
-     * @param array  $haystack array to searched
-     */
-    static function str_remove_deep($needle, &$haystack) {
-        array_walk_recursive($haystack,
-          function (&$h, &$k) use ($needle) {
-              $h = str_replace($needle, '', $h);
-        });
-    }
-
-    /**
-     * Replaces line breaks by DokuWiki's \\ line breaks and inserts
-     * <nowiki> tags.
-     *
-     * @param string $text
-     * @return string
-     */
-    function textAsWiki($text) {
-        // First, remove existing </nowiki> end tags. (We display events that contain '</nowiki>'
-        // incorrectly but this should not be a problem in practice.)
-        // Second, replace line breaks by DokuWiki line breaks.
-        $needle   = array('</nowiki>', "\n");
-        $haystack = array('',          $this->nowikiEnd() . '\\\\ '. $this->nowikiStart());
-        $text = str_ireplace($needle, $haystack, $text);
-        return $this->nowikiStart() . $text . $this->nowikiEnd();
-    }
-
-    function magicString() {
-        return '{' . $this->nonce .' magiccc}';
+        $renderer->doc .= static::ERROR_PREFIX . 'The plugin requires at least PHP 5.5.';
+        return false;
     }
+}
 
-    function nowikiStart() {
-        return '<nowiki>' . $this->magicString();
-    }
 
-    function nowikiEnd() {
-        return $this->magicString() . '</nowiki>';
+// An 'require' ensures that older PHP versions do not even try to parse the actual code.
+if (version_compare(PHP_VERSION, '5.5.0') >= 0) {
+    require 'syntax-impl.php';
```

Comment:
use `__DIR__` to require from absolute path, not from php include path (generic name might hit something)

also, compare `PHP_VERSION_ID` instead. `PHP_VERSION_ID` is available as of PHP 5.2.7.